lighthouse-core/test/audits/treemap-data-test.js

```diff
@@ -0,0 +1,335 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const TreemapData_ = require('../../audits/treemap-data.js');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
+const {loadSourceMapAndUsageFixture, makeParamsOptional} = require('../test-utils.js');
+
+/* eslint-env jest */
+
+const TreemapData = {
+  audit: makeParamsOptional(TreemapData_.audit),
+  prepareTreemapNodes: makeParamsOptional(TreemapData_.prepareTreemapNodes),
+};
+
+/**
+ * @param {string} url
+ * @param {number} resourceSize
+ * @param {LH.Crdp.Network.ResourceType} resourceType
+ */
+function generateRecord(url, resourceSize, resourceType) {
+  return {url, resourceSize, resourceType};
+}
+
+describe('TreemapData audit', () => {
+  describe('squoosh fixture', () => {
+    /** @type {import('../../audits/treemap-data.js').TreemapData} */
+    let treemapData;
+    beforeAll(async () => {
+      const context = {computedCache: new Map()};
+      const {map, content, usage} = loadSourceMapAndUsageFixture('squoosh');
+      const mainUrl = 'https://squoosh.app';
+      const scriptUrl = 'https://squoosh.app/main-app.js';
+      const networkRecords = [generateRecord(scriptUrl, content.length, 'Script')];
+
+      // Add a script with no source map or usage.
+      const noSourceMapScript = {src: 'https://sqoosh.app/no-map-or-usage.js', content: '// hi'};
+      networkRecords.push(
+        generateRecord(noSourceMapScript.src, noSourceMapScript.content.length, 'Script')
+      );
+
+      const artifacts = {
+        URL: {requestedUrl: mainUrl, finalUrl: mainUrl},
+        JsUsage: {[usage.url]: [usage]},
+        devtoolsLogs: {defaultPass: networkRecordsToDevtoolsLog(networkRecords)},
+        SourceMaps: [{scriptUrl: scriptUrl, map}],
+        ScriptElements: [{src: scriptUrl, content}, noSourceMapScript],
+      };
+      const results = await TreemapData.audit(artifacts, context);
+
+      // @ts-expect-error: Debug data.
+      treemapData = results.details.treemapData;
+    });
+
+    it('basics', () => {
+      expect(Object.keys(treemapData)).toEqual(['scripts', 'resources']);
+    });
+
+    it('scripts', () => {
+      expect(treemapData.scripts.find(s => s.name === 'https://sqoosh.app/no-map-or-usage.js'))
+        .toMatchInlineSnapshot(`
+        Object {
+          "name": "https://sqoosh.app/no-map-or-usage.js",
+          "node": Object {
+            "name": "https://sqoosh.app/no-map-or-usage.js",
+            "resourceBytes": 37,
+          },
+        }
+      `);
+
+      expect(treemapData.scripts).toMatchSnapshot();
```

Comment:
given that github collapses changes to that file by default I'm a little less comfortable with that as our only test (seems likely reviewers aren't going to fully understand and catch changes to that file).

maybe we could get some more specific assertions using `toMatchObject` instead?

Reply:
and maybe add back in `expect(JSON.stringify(treeMapData.scripts).length).toMatchInlineSnapshot()` if we're concerned about catching *any* changes to this?